developer_guide.md

```diff
@@ -0,0 +1,43 @@
+
+# Developer Guide
+
+## Git env
```

Comment:
I guess this is just standard github workflow, not sure how useful to include it here.

Reply:
google "github sync with upstream", top answers are using `git merge`, it would be nice to help new dev follow best practice of git.

Reply:
@rongou thanks for code review, all others are fixed.